### PR TITLE
Change default pvc to 10G.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this PR do?
https://github.com/eclipse/che/issues/19314

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/master/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/master/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

**Review by mmorhun — APPROVED** [review]

File: pkg/controller/che/che_controller.go [hunk 1]
```diff
@@ -704,14 +705,36 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
...
 
 	if cheMultiUser == "false" {
-		done, err := deploy.SyncPVCToCluster(deployContext, deploy.DefaultCheVolumeClaimName, "1Gi", cheFlavor)
+		claimSize := util.GetValue(instance.Spec.Storage.PvcClaimSize, deploy.DefaultPvcClaimSize)
+		done, err := deploy.SyncPVCToCluster(deployContext, deploy.DefaultCheVolumeClaimName, claimSize, cheFlavor)
 		if !tests {
 			if !done {
 				logrus.Infof("Waiting on pvc '%s' to be bound. Sometimes PVC can be bound only when the first consumer is created.", deploy.DefaultCheVolumeClaimName)
-				if err != nil {
+				if err != nil && strings.HasSuffix(err.Error(), pvcIsNotResizableError) {
```
Comment (tolusha) [inline pkg/controller/che/che_controller.go:713]: Pls. Just log error message without falling back to default pvcClaim size value.

File: pkg/controller/che/che_controller.go [hunk 2]
```diff
@@ -704,15 +704,15 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 	cheMultiUser := deploy.GetCheMultiUser(instance)
 
 	if cheMultiUser == "false" {
-		done, err := deploy.SyncPVCToCluster(deployContext, deploy.DefaultCheVolumeClaimName, "1Gi", cheFlavor)
-		if !tests {
-			if !done {
+		claimSize := util.GetValue(instance.Spec.Storage.PvcClaimSize, deploy.DefaultPvcClaimSize)
+		done, err := deploy.SyncPVCToCluster(deployContext, deploy.DefaultCheVolumeClaimName, claimSize, cheFlavor)
+		if !tests && !done {
```
Comment (tolusha) [inline pkg/controller/che/che_controller.go:709]:
```suggestion
		if !done {
```

File: pkg/controller/che/create.go [hunk 3]
```diff
@@ -173,13 +173,6 @@ func (r *ReconcileChe) GenerateAndSaveFields(deployContext *deploy.DeployContext
 			return err
 		}
 	}
-	pvcClaimSize := util.GetValue(deployContext.CheCluster.Spec.Storage.PvcClaimSize, deploy.DefaultPvcClaimSize)
-	if len(deployContext.CheCluster.Spec.Storage.PvcClaimSize) < 1 {
-		deployContext.CheCluster.Spec.Storage.PvcClaimSize = pvcClaimSize
-		if err := r.UpdateCheCRSpec(deployContext.CheCluster, "pvc claim size", pvcClaimSize); err != nil {
-			return err
-		}
-	}
```
Comment (tolusha) [inline pkg/controller/che/create.go:182]: Pls revert

**Review by tolusha — APPROVED** [review]

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
ci/prow/v7-devworkspace-happy-path is broken, probably related to https://github.com/che-incubator/devworkspace-che-operator/pull/33
```ime="2021-04-02T09:47:35Z" level=info msg="Creating a new object: CustomResourceDefinition, name: chemanagers.che.eclipse.org"
time="2021-04-02T09:47:35Z" level=error msg="CustomResourceDefinition.apiextensions.k8s.io \"chemanagers.che.eclipse.org\" is invalid: spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required"
time="2021-04-02T09:47:36Z" level=info msg="Added finalizer: oauthclients.finalizers.che.eclipse.org"
time="2021-04-02T09:47:36Z" level=info msg="Removing 'cheWorkspaces.clusterpermissions.finalizers.che.eclipse.org'"
time="2021-04-02T09:47:36Z" level=info msg="Deleted finalizer: cheWorkspaces.clusterpermissions.finalizers.che.eclipse.org"
time="2021-04-02T09:47:44Z" level=info msg="Creating a new object: CustomResourceDefinition, name: chemanagers.che.eclipse.org"
time="2021-04-02T09:47:44Z" level=error msg="CustomResourceDefinition.apiextensions.k8s.io \"chemanagers.che.eclipse.org\" is invalid: spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required"
time="2021-04-02T09:47:52Z" level=info msg="Creating a new object: CustomResourceDefinition, name: chemanagers.che.eclipse.org"
time="2021-04-02T09:47:52Z" level=error msg="CustomResourceDefinition.apiextensions.k8s.io \"chemanagers.che.eclipse.org\" is invalid: spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required"
time="2021-04-02T09:48:01Z" level=info msg="Creating a new object: CustomResourceDefinition, name: chemanagers.che.eclipse.org"
time="2021-04-02T09:48:01Z" level=error msg="CustomResourceDefinition.apiextensions.k8s.io \"chemanagers.che.eclipse.org\" is invalid: spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required"
time="2021-04-02T09:48:10Z" level=info msg="Creating a new object: CustomResourceDefinition, name: chemanagers.che.eclipse.org"
time="2021-04-02T09:48:10Z" level=error msg="CustomResourceDefinition.apiextensions.k8s.io \"chemanagers.che.eclipse.org\" is invalid: spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required"
time="2021-04-02T09:48:18Z" level=info msg="Creating a new object: CustomResourceDefinition, name: chemanagers.che.eclipse.org"
time="2021-04-02T09:48:18Z" level=error msg="CustomResourceDefinition.apiextensions.k8s.io \"chemanagers.che.eclipse.org\" is invalid: spec.versions[0].schema.openAPIV3Schema: Required value: schemas are required"
time="2021-04-02T09:48:27Z" level=info msg="Creating a new object: CustomResourceDefinition, name: chemanagers.che.eclipse.org"
```

**Pull request merged**